### PR TITLE
Add element weighting parameter to BatchElements.

### PR DESCRIPTION
**Pull request by robertwb**
This is useful when various elements have largely disparate weights.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @ryanthompson591

**Comment by robertwb** [comment]
Irrelevant failure in apache_beam.transforms.userstate_test.StatefulDoFnOnDirectRunnerTest.test_dynamic_timer_clear_then_set_timer

**Comment by robertwb** [comment]
Run Python PreCommit

File: sdks/python/apache_beam/transforms/util_test.py [hunk 1]
```diff
@@ -228,6 +228,16 @@ def test_windowed_batches(self):
               7,  # elements in [30, 47)
           ]))
 
+  def test_sized_batches(self):
```
Comment (ryanthompson591) [inline sdks/python/apache_beam/transforms/util_test.py:231]: I think when someone looks at this in the future and says, "why is this here", you could illustrate it better with a test like this (I'm not sure if this works but you get the point).

def test_element_size_fn()
""" Tests passing different sized arrays as elements and then weighting them by how large the arrays are. Batch sizes will depend on the size of the arrays within."""
...
| beam.Create([[1]*1, [1] *1, [10] *10, [6] * 6 ...], reshuffle=False)
| util.BatchElements(min_batch_size =10, max_batch_size=10, element_size_fn=lambda x: len(x))
| beam.Map(len)
assert_that(res, equal_to([12, 11, 10]))
...

I think that might just make it clearer to the future reader as to why this feature might exist.  Feel free to take or ignore this suggestion.

**Review by ryanthompson591 — APPROVED** [review]

File: sdks/python/apache_beam/transforms/util.py [hunk 2]
```diff
@@ -632,6 +649,8 @@ def __init__(
       max_batch_size=10000,
       target_batch_overhead=.05,
       target_batch_duration_secs=1,
+      *,
```
[the remark this answers is not in the document]
Comment (robertwb) [reply sdks/python/apache_beam/transforms/util.py:652]: Forces these later ones to be keyword arguments (so their order doesn't matter).

File: sdks/python/apache_beam/transforms/util_test.py (hunk 1 above)
Comment (robertwb) [reply sdks/python/apache_beam/transforms/util_test.py:231]: Good idea. The mix of `[]`'s and `*`'s are a bit hard for me to read, so I'm using strings.

**Pull request merged**